### PR TITLE
build: bump Angular dependencies and drop benchpress

### PR DESCRIPTION
**Pull request by crisbeto**
* Updates the `@angular` dependencies to the latest versions and resolves the build errors.
* Removes the dependency on `@angular/benchpress` since we weren't using it and it was causing some issues in the snapshot builds.

### CODE REVIEW

**Review by devversion — APPROVED** [review]
LGTM

**Review by josephperrott — CHANGES_REQUESTED** [review]
An issue with validating commits was found for when multiple level scopes as used and will be fixed with: https://github.com/angular/angular/pull/41430#pullrequestreview-627042875

The dev-infra-private-builds version needs to include this change.

File: package.json [hunk 1]
```diff
@@ -71,16 +71,15 @@
...
-    "@angular/dev-infra-private": "https://github.com/angular/dev-infra-private-builds.git#a8dbb0d8cd8de1e86a4ec4ea66b5bd89a47d4876",
-    "@angular/platform-browser-dynamic": "^12.0.0-next.5",
-    "@angular/platform-server": "^12.0.0-next.5",
-    "@angular/router": "^12.0.0-next.5",
+    "@angular-devkit/build-optimizer": "^0.1200.0-next.6",
+    "@angular-devkit/core": "^12.0.0-next.6",
+    "@angular-devkit/schematics": "^12.0.0-next.6",
+    "@angular/bazel": "^12.0.0-next.7",
+    "@angular/compiler-cli": "^12.0.0-next.7",
+    "@angular/dev-infra-private": "https://github.com/angular/dev-infra-private-builds.git#ebf0a487a4de71746b5b025e4e7878ca5714ba46",
```
Comment (josephperrott) [inline package.json:79]:
```suggestion
    "@angular/dev-infra-private": "https://github.com/angular/dev-infra-private-builds.git#3953ba9eea35a2660b6068523bafb92634042be9",
```

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by annieyw** [comment]
@crisbeto  needs rebase

**Comment by crisbeto** [comment]
Rebased.

**Pull request merged**